### PR TITLE
python3Packages.dunamai: init at 1.10.0

### PR DESCRIPTION
**Pull request by jmgilman**
###### Description of changes

Adds the [dunamai](https://github.com/mtkennerly/dunamai) python package. 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/dunamai/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+, pythonOlder
+, importlib-metadata
+, packaging
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "dunamai";
+  version = "1.10.0";
+  format = "pyproject";
```
Comment (fabaff) [inline pkgs/development/python-modules/dunamai/default.nix:14]: Only Python > 3.5 is [supported](https://github.com/mtkennerly/dunamai/blob/master/pyproject.toml#L20). As all 
 Python releases < 3.7 are EOL, I suggest to set it 3.7 (can always be adjusted if one confirm that the module works.). 

```suggestion
  format = "pyproject";

  disabled = pythonOlder "3.7";
```

**Review by fabaff — COMMENTED** [review]
Can you please squash the commit? But keep the one for `maintainer-list.nix`.

File: pkgs/development/python-modules/dunamai/default.nix (hunk 1 above)
Comment (jmgilman) [reply pkgs/development/python-modules/dunamai/default.nix:14]: Is it pretty standard to enforce minimum versions by disabling the package? Just trying to learn proper decorum for my future contributions :) Thanks for the tip.

**Comment by jmgilman** [comment]
> Can you please squash the commit? But keep the one for `maintainer-list.nix`.

@fabaff Done.

**Comment by fabaff** [comment]
The commit message for `maintainer-list.nix` should not mention a Python module. Could you do a `git commit --amend`?

```text
maintainers: add @jmgilman
```

**Comment by jmgilman** [comment]
> The commit message for `maintainer-list.nix` should not mention a Python module. Could you do a `git commit --amend`?
> 
> ```
> maintainers: add @jmgilman
> ```

Done.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 164784` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.dunamai</li>
    <li>python39Packages.dunamai</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/164784#issuecomment-1075077344)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-164784]$ ./results/python310Packages.dunamai/bin/dunamai 
usage: dunamai [-h] {from,check} ...
dunamai: error: the following arguments are required: command
```

Looks good to me.

**Comment by fabaff** [comment]
Thanks @jmgilman for your first package.

**Pull request merged**